rbd/snapshot_rename.go

```diff
@@ -0,0 +1,38 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <stdlib.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+// Rename a snapshot.
+//  PREVIEW
+//
+// Implements:
+// 	int rbd_snap_rename(rbd_image_t image, const char *snapname,
+//				 const char* dstsnapsname);
+func (snapshot *Snapshot) Rename(destName string) error {
+	if err := snapshot.validate(imageNeedsIOContext | imageIsOpen | imageNeedsName | snapshotNeedsName); err != nil {
+		return err
+	}
+
+	cSrcName := C.CString(snapshot.name)
+	cDestName := C.CString(destName)
+	defer C.free(unsafe.Pointer(cSrcName))
+	defer C.free(unsafe.Pointer(cDestName))
+
+	err := C.rbd_snap_rename(snapshot.image.image, cSrcName, cDestName)
+	if err != 0 {
+		return getError(err)
+	}
+
+	snapshot.name = destName
```

Comment:
Yeah. Interesting point. I've always felt that the rbd snapshot api was awkward/suboptimal since becoming invovled with go-ceph. This is another case of that.  We can somewhat handwave the problem away by saying this snapshot name is just the locally cached version... but I do think it could be an issue and why a "thick" API can go wrong. :-)

Reply:
Sure, we can cache, but caches need to time out at least, and even better get invalidated by watchers. ;-) Anyway, I'm not so serious here in this specific case, but this stuff can definitely lead to confusion.